### PR TITLE
mon: show pool id in pool ls command

### PR DESCRIPTION
**Pull request by liuchang0812**
Signed-off-by: Chang Liu <liuchang0812@gmail.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

```
build git:(show-pool-id-in-pool-ls-cmd) ✗ ./bin/ceph osd pool ls detail -f json-pretty
[
    {
        "poolid": 1,
        "pool_name": "reppool",
        "create_time": "2019-06-11T19:21:16.126324+0800",
        "flags": 1,
        "flags_names": "hashpspool",

```

### CODE REVIEW

**Comment by liuchang0812** [comment]
> I think "id" would be preferred and is at least consistent with `dump_pool_stats_full()`.

thank you. I found we use `poolid` in `pg dump` command.

```
(env) ➜  build git:(mgr-progress-fixes) ✗ ./bin/ceph pg dump -f json-pretty 2>/dev/null | grep "poolid" -B 3
        ],
        "pool_stats": [
            {
                "poolid": 1,
```

**Comment by batrick** [comment]
@neha-ojha  or @liewegas can you add this to your next RADOS run.

**Comment by batrick** [comment]
retest this please

**Comment by neha-ojha** [comment]
https://tracker.ceph.com/issues/40370

**Comment by liewegas** [comment]
Should we make the simple output mode do a TextTable with an ID column?

**Comment by liuchang0812** [comment]
> Should we make the simple output mode do a TextTable with an ID column?

how about ?
```
./bin/ceph osd pool ls detail  2>/dev/null
    ID     NAME        TYPE     SIZE     MIN_SIZE     CRUSH_RULE     OBJECT_HASH     PG_NUM     PGP_NUM     AUTUSCALE_MODE     LAST_CHANGE     FLAGS      
     1      ecpool        3        3            2              1        rjenkins         16          16               warn              15     hashpspool 
     2     reppool        1        3            1              0        rjenkins         16          16               warn              18     hashpspool 
```

**Comment by jecluis** [comment]
As a table, I'd rather show the `TYPE` as a string instead of the type's id. While JSON is to be consumed by whatever, we need text to be consumable by a human - and humans don't typically know that a pool type 3 refers to an ec pool ;)

And I would even do the same for the crush rule column. The same rationale; or just drop it - I don't see much value in showing the crush rule for a pool in its details.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-16_17:08:46-rados-wip-kefu-testing-2019-06-16-2148-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-06-18_09:55:35-rados-wip-kefu-testing-2019-06-16-2148-distro-basic-smithi/

failures already tracked/addressed by

- http://tracker.ceph.com/issues/17882
- 2dd5574cf9a5a80cc40a365675bb1e33a32c9f41

**Comment by liuchang0812** [comment]
@tchaikov could we merge this PR now? I will create another PR to make this command output with a TextTable formatter.

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -5296,6 +5296,7 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
 	if (f) {
 	  if (detail == "detail") {
 	    f->open_object_section("pool");
+	    f->dump_int("poolid", it->first);
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: nit: can we make this pool_id for consistency?

**Comment by liewegas** [comment]
> > Should we make the simple output mode do a TextTable with an ID column?
> 
> how about ?
> 
> ```
> ./bin/ceph osd pool ls detail  2>/dev/null
>     ID     NAME        TYPE     SIZE     MIN_SIZE     CRUSH_RULE     OBJECT_HASH     PG_NUM     PGP_NUM     AUTUSCALE_MODE     LAST_CHANGE     FLAGS      
>      1      ecpool        3        3            2              1        rjenkins         16          16               warn              15     hashpspool 
>      2     reppool        1        3            1              0        rjenkins         16          16               warn              18     hashpspool 
> ```

- use type string
- translate crush rule to name
- drop object_hash, last_change
- left justify name

...probably in a separate pr?

**Comment by liuchang0812** [comment]
> > > Should we make the simple output mode do a TextTable with an ID column?
> > 
> > 
> > how about ?
> > ```
> > ./bin/ceph osd pool ls detail  2>/dev/null
> >     ID     NAME        TYPE     SIZE     MIN_SIZE     CRUSH_RULE     OBJECT_HASH     PG_NUM     PGP_NUM     AUTUSCALE_MODE     LAST_CHANGE     FLAGS      
> >      1      ecpool        3        3            2              1        rjenkins         16          16               warn              15     hashpspool 
> >      2     reppool        1        3            1              0        rjenkins         16          16               warn              18     hashpspool 
> > ```
> 
> * use type string
> * translate crush rule to name
> * drop object_hash, last_change
> * left justify name
> 
> ...probably in a separate pr?

thank you for your advice, Sage I will implement those in a separate PR.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-08-21_12:58:47-rados-wip-kefu-testing-2019-08-21-1445-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-08-23_00:12:22-rados-wip-kefu-testing-2019-08-21-1445-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-08-23_03:12:13-rados-wip-kefu-testing-2019-08-21-1445-distro-basic-smithi/

**Pull request merged**